torch/_dynamo/decorators.py

```diff
@@ -575,34 +575,47 @@ def mark_unbacked(
         specialize_on (Optional[list[Any]], default=None): A list of specialization criteria (e.g., lambdas) for this dimension.
             If provided, Dynamo will generate specialized compiled regions for each criterion in addition to a generic trace.
     """
-    # You could have copied the mark_dynamic behavior but I'm not convinced
-    # it's what you want
-    assert not is_traceable_wrapper_subclass(t), "not implemented yet"
+    if isinstance(t, torch.distributed.tensor.DTensor):
+        # apply on inner tensor sizes/strides
+        _apply_func_to_inner_tensors_of_same_dim(mark_unbacked, t, index)
```

Comment:
we only need to handle _local_tensor its more clear to just do 
mark_unbacked(t._local_tensor, index)